homeassistant/components/group/fan.py

```diff
@@ -151,6 +172,13 @@ def async_update_supported_features(
                     self._fans[feature].add(entity_id)
                 else:
                     self._fans[feature].discard(entity_id)
+            preset_modes = new_state.attributes.get(ATTR_PRESET_MODES)
```

Comment:
This is only handling the last entity_id/state it sees.  We need to store the preset modes in `self.??` and then build `self._preset_modes` from only the ones that all the entities share.

Reply:
I don't have other fans to test my theory.
Is there perhaps some dummy fan available in home assistant with different preset_modes for testing such scenario ?

Reply:
I just tested how it work with debugger and it seams, that it remebers self._preset_modes in callback.

I'll prepare video for this :)

Reply:
Video uploaded.

It starts on first run breakpoint when it reads first fan in the group.
self._preset_modes is None.
Then self._preset_modes get values of first fan.

In next iteration value of self._preset_modes is alredy seted. So else: get excecuted and intersection between self._preset_modes  and new read preset_modes from current fan is made.